core/src/main/java/apoc/nodes/Grouping.java

```diff
@@ -368,6 +377,10 @@ private <C extends Collection<T>, T extends Entity> C fixAggregates(C pcs) {
                     double[] values = (double[]) v;
                     entry.setValue(values[1] == 0 ? 0 : values[0] / values[1]);
                 }
+                if (k.matches("^avg_.+") && v instanceof DurationValue) {
```

Comment:
Regexing the property name, not very rubust, but I guess this is the way of the apoc.

Reply:
Compiling a regex pattern is a bit expensive though (I don't think they're cached in java) so we could consider replacing this with `k.startsWith("avg_") && k.size() > "avg_".length`. Or to compile the Pattern once in a field and reuse.